### PR TITLE
auth: pdnsutil zonemd-verify-file tests from github.com/verisign/zonemd-test-cases

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

This adds the test from github.com/verisign/zonemd-test-cases, with a few small modifications, as
explained in the respective commits.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [X] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: regression-tests.nobackend/zonemd-test-cases/zones/50-uppercase-nsec-rdata-names/arpa.zone.hashed [hunk 1]
```diff
@@ -1,3 +1,4 @@
+arpa.	86400	IN	SOA	a.root-servers.net. nstld.verisign-grs.com. 2021062901 1800 900 604800 86400
```
Comment (Habbie) [inline regression-tests.nobackend/zonemd-test-cases/zones/50-uppercase-nsec-rdata-names/arpa.zone.hashed:1]: do we require SOA to be moved up here?

File: regression-tests.nobackend/zonemd-test-cases/zones/51-uppercase-nsec3-rdata-names/arpa.zone.hashed [hunk 2]
```diff
@@ -15,17 +15,17 @@ arpa.	518400	IN	RRSIG	NS 8 1 518400 20210616170429 20210519170429 29094 arpa. gy
...
 arpa.	86400	IN	DNSKEY	257 3 8 AwEAAdQP1t2ookuQYFNUNGDmLHcoA6LFSImvULaUgChKiIO6Vv5yDyHB0Ng6ZkfHM0586cLcbXNBLj/9u5A4vqzOFj8phzW4WLZREZBLYMcuHhvQdqzuDJ0J5mxmLLis5eNaCwukVm6Zpf/otzCJsx9LyrhQBTyx6FF+h7dbSCvjh7tD ;{id = 18949 (ksk), size = 1024b}
 arpa.	3600	IN	NSEC3PARAM	1 0 1 - 
-arpa.	86400	IN	TYPE63	\# 54 7876cdfe01019a84145013e13e3de2328868888c65aa46b7381213990f83d496c642d2324029cc852e09bffa38afd8e9197977776591
+arpa.	86400	IN	ZONEMD	2021051902 1 1 9a84145013e13e3de2328868888c65aa46b7381213990f83d496c642d2324029cc852e09bffa38afd8e9197977776591
 0js82oec35lbbc4hl35476cm5icacksf.arpa.	86400	IN	RRSIG	NSEC3 8 2 86400 20210616170429 20210519170429 29094 arpa. PRVkH4+Nm17QlFgwFLnoqwaiIwWZ4pvscanHdMb6HOKkSxwtDoWAGhZubvYGt/Je735nQkGQPPXW2tkMkJa3D7e6RkX/8AoxcqqXOimC6BlG6LuSL4rSousDlbrulyh87qgIHXkUtrHyYUNAMZMKOjMHo7t5IxwjBO0SGADoglk=
-0js82oec35lbbc4hl35476cm5icacksf.arpa. 86400 IN NSEC3 1 0 1 - 2UB8EN7BK0T6DENIGO3I729IVQVME3VE NS
+0js82oec35lbbc4hl35476cm5icacksf.arpa.	86400	IN	NSEC3	1 0 1 -  2ub8en7bk0t6denigo3i729ivqvme3ve NS
```
Comment (Habbie) [inline regression-tests.nobackend/zonemd-test-cases/zones/51-uppercase-nsec3-rdata-names/arpa.zone.hashed:24]: do we need the lowercasing?

**Review by Habbie — APPROVED** [review]
Two questions (to which I suspect the answer is "that's what happened after `ldns-read-zone`" or something). Otherwise, looks good!

File: regression-tests.nobackend/zonemd-test-cases/zones/50-uppercase-nsec-rdata-names/arpa.zone.hashed (hunk 1 above)
Comment (omoerbeek) [reply regression-tests.nobackend/zonemd-test-cases/zones/50-uppercase-nsec-rdata-names/arpa.zone.hashed:1]: no, and you gave the answer already :-)

File: regression-tests.nobackend/zonemd-test-cases/zones/51-uppercase-nsec3-rdata-names/arpa.zone.hashed (hunk 2 above)
Comment (omoerbeek) [reply regression-tests.nobackend/zonemd-test-cases/zones/51-uppercase-nsec3-rdata-names/arpa.zone.hashed:24]: no, and you gave the answer already :-)

**Comment by omoerbeek** [comment]
I'll redo the fixes to the test files to please our parser in a minimal way.

**Review by Habbie — APPROVED** [review]

**Pull request merged**